pal_event_server/lib/src/loggers/app_usage/macos/macos_helper.dart

```diff
@@ -30,21 +30,24 @@ void macOSAppLoggerIsolate(SendPort sendPort) {
   Timer.periodic(queryInterval, (Timer _) {
     Process.run(apple_script.command, apple_script.scriptArgs).then((result) {
       final currWindow = result.stdout.trim();
-      final resultMap = apple_script.buildResultMap(currWindow);
-      final currAppAndWindowName =
-          resultMap[appNameField] + resultMap[windowNameField];
-
-      if (currAppAndWindowName != _prevAppAndWindowName) {
-        // Send APP_CLOSED
-        if (_prevAppAndWindowName != null && _prevAppAndWindowName.isNotEmpty) {
-          sendPort.send(_prevAppAndWindowName);
-        }
-
-        _prevAppAndWindowName = currAppAndWindowName;
-
-        // Send PacoEvent && APP_USAGE
-        if (resultMap != null) {
-          sendPort.send(resultMap);
+      if (currWindow != '') {
```

Comment:
It looks like this changes the app change detection logic? If we get currWindow == '' then that represents a potentially new window, we just don't know which one. So, we still need to track it as a new window.

Reply:
I have played with the script a few times and it appears that currWindow=='' usually means the applescript failed with an error. When there is a new window, there is usually at least an app name. And since we query the window every second, if we actually switched to a new window, the switch will be captured in the next few cycles.